### PR TITLE
Add support for dictionary items

### PR DESCRIPTION
**Pull request by icco**
This is an attempt to fix the issues raised in #70 to get it merged into the gem.

### CODE REVIEW

**Comment by aspires** [comment]
@fastly/api por favor

**Comment by lanej** [comment]
@icco it seems the method signatures for `delete_path` and `put_path` have mixed expectations of `Hash` vs `Fastly::DictionaryItem`

```ruby
irb(main):018:0> dict.delete_item('key_a')
NoMethodError: undefined method `service_id' for #<Hash:0x007f962b9d27b8>
	from /Users/jlane/p/fastly-ruby/lib/fastly/dictionary_item.rb:29:in `put_path'
	from /Users/jlane/p/fastly-ruby/lib/fastly/dictionary_item.rb:33:in `delete_path'
```

**Comment by icco** [comment]
@lanej fixed. Not the greatest, but open to suggestions on the best solutions.

**Comment by icco** [comment]
@fastly/api ping

File: lib/fastly/dictionary.rb [hunk 1]
```diff
@@ -2,6 +2,28 @@ class Fastly
...
+      if di
+        di.item_value = value
+        fetcher.update_dictionary_item(di)
+      else
+        add_item(key, value)
+      end
+    end
+
+    def delete_item(key)
+      fetcher.delete_dictionary_item(service_id: service_id, dictionary_id: id, item_key: key)
```
Comment (lanej) [inline lib/fastly/dictionary.rb]: this still uses the 'options` interface resulting in :

```ruby
irb(main):016:0> pp dict.delete_item('key_a')
NoMethodError: undefined method `service_id' for #<Hash:0x007fcb0d83fb78>
	from /Users/jlane/fastly/fastly-ruby/lib/fastly/dictionary_item.rb:33:in `delete_path'
	from /Users/jlane/fastly/fastly-ruby/lib/fastly/fetcher.rb:48:in `delete'
	from /Users/jlane/fastly/fastly-ruby/lib/fastly.rb:171:in `block (2 levels) in <class:Fastly>'
	from /Users/jlane/fastly/fastly-ruby/lib/fastly/dictionary.rb:24:in `delete_item'
```

Comment (icco) [inline lib/fastly/dictionary.rb]: Ah, I misunderstood what was going on sorry. Fixed.

```
2.3.0 :008 > edge_dict.delete_item "/from"
 => true
```

**Comment by icco** [comment]
Ok, sorry about that @aspires or @lanej. Now it's fixed. Tested and everything.

**Comment by lanej** [comment]
Thanks @icco

**Comment by icco** [comment]
Thank you!

**Pull request merged**